src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -536,27 +527,33 @@ result::Download SotaUptaneClient::downloadImages(const std::vector<Uptane::Targ
   result::Download result;
   std::vector<Uptane::Target> downloaded_targets;
 
+  result::UpdateStatus update_status = checkUpdatesOffline(targets);
```

Comment:
@patrickvacek - this looks to be the call that's breaking aktualizr-lite. The problem is that downloadImages was originally a code path that did not have any Uptane dependencies and `checkUpdatesOffline` is adding one in.

I haven't fully thought through this, but one way we could avoid the Uptane path is by making the `downloadImage` method public so that aktualizr-lite could call that directly and maybe the new `pacman->verifyTarget` API as well?

Reply:
> this looks to be the call that's breaking aktualizr-lite.

Agreed.

> The problem is that downloadImages was originally a code path that did not have any Uptane dependencies and `checkUpdatesOffline` is adding one in.

Yes, it should've always been there, but we were lazy before.

> one way we could avoid the Uptane path is by making the `downloadImage` method public so that aktualizr-lite could call that directly

Not my favorite solution, but I'm not sure how many options we have, and I think we've already set the precedent that it's okay for tools in the aktualizr repo to access SotaUptaneClient functions directly, but "normal" users should go through the Aktualizr class API.

> maybe the new `pacman->verifyTarget` API as well

That is called automatically during installation. Why would you want to call it after downloading?

> We could probably add a verifyTarget method to the docker-app-manager code that checks to validate the .dockerapp target is syntactically valid.

Cool, sounds like a good idea!

> I'm traveling to ELC today, so it might be a couple of days before I get real time to debug.

That's okay, I still have some work to do on this PR and it's been slow progress due to competing priorities. Anyway, have fun!

Reply:
> > maybe the new `pacman->verifyTarget` API as well
> 
> That is called automatically during installation. Why would you want to call it after downloading?

Now I understand; it's because you don't call `uptaneInstall` directly!

I wish it were possible for you to use the same API but just skip the parts you don't need for your package manager of choice. Is it also possible to mock the DockerApp targets like we do for the fake package manager?